checker/src/org/checkerframework/checker/formatter/FormatterAnnotatedTypeFactory.java

```diff
@@ -147,5 +151,51 @@ public boolean isSubtype(AnnotationMirror rhs, AnnotationMirror lhs) {
 
             return super.isSubtype(rhs, lhs);
         }
+
+        @Override
+        public AnnotationMirror greatestLowerBound(AnnotationMirror anno1,
+                AnnotationMirror anno2) {
+            if (AnnotationUtils.areSameIgnoringValues(anno1, FORMAT) &&
+                AnnotationUtils.areSameIgnoringValues(anno2, FORMAT)) {
+                ConversionCategory[] anno1ArgTypes =
+                        treeUtil.formatAnnotationToCategories(anno1);
+                ConversionCategory[] anno2ArgTypes =
+                        treeUtil.formatAnnotationToCategories(anno2);
+
+                if (anno1ArgTypes == null && anno2ArgTypes == null) {
+                	return FORMAT;
+                }
+
+                if (anno1ArgTypes == null || anno2ArgTypes == null) {
+                    return FORMATBOTTOM; // TODO: Should this be InvalidFormat instead?
```

Comment:
I am not sure about this and the case below. Perhaps Konstantin should also have a look as he is probably most familiar with this type system.